src/main/java/lotto/domain/winning/WinningNumbers.java

```diff
@@ -5,30 +5,41 @@
 import lotto.domain.grade.Grade;
 import lotto.domain.lotto.Lotto;
 import lotto.domain.lotto.LottoNumber;
+import lotto.error.exception.SizeExceedLottoException;
 
 public class WinningNumbers {
 
     private final List<LottoNumber> winningNumbers;
 
-    public WinningNumbers(List<LottoNumber> winningNumbers) {
+    private final LottoNumber bonusNumber;
+
+    public WinningNumbers(List<LottoNumber> winningNumbers, LottoNumber bonusNumber) {
         this.winningNumbers = winningNumbers;
+        this.bonusNumber = bonusNumber;
     }
 
-    public static WinningNumbers fromValues(final List<Integer> values) {
+    public static WinningNumbers fromValues(final List<Integer> values, LottoNumber bonusNumber) {
+        if (values.size() != 6) {
+            throw new SizeExceedLottoException(values.size());
+        }
+
         List<LottoNumber> lottoNumbers = values.stream()
             .map(LottoNumber::fromInt)
             .collect(Collectors.toList());
-        return new WinningNumbers(lottoNumbers);
+        return new WinningNumbers(lottoNumbers, bonusNumber);
     }
 
     public Grade confirmWinning(final Lotto lotto) {
-        return Grade.fromCorrectingCount(findNumberThatMatchWinningBall(lotto));
+        return Grade.fromMatchResult(findMatchResultByLotto(lotto));
```

Comment:
Nit: `WinningNumberResult`로 감싸서 처리하신 이유가 있으실까요?
개인적으로는 일치하는 번호의 개수와 보너스 번호 유무를 곧바로 직접 전달해도 되지 않나란 생각이 듭니다.
```suggestion
        int matchCount = calculateMatchingLottoNumbers(lotto);
        boolean hasBonusNumber = lotto.contains(bonusNumber);
        return Grade.from(matchCount, hasBonusNumber);
```

Reply:
WinningNumberResult로 감쌀 경우 isBonusGrade와 같은 메소드 책임을 부여할 수도 있고 matchCount를 구하는 경우와 bonusNumber를 구하는 역할을 분리해서 한 가지 역할만 수행할 수 있을거라고 생각했었거든요 

그런데 선동님 피드백 주신 부분 보니까 너무 감싼거 아닌가라는 생각이 들기도 하네요. 이 부분도 피드백 반영해서 step4에 반영하겠습니다.